src/coreclr/vm/method.inl

```diff
@@ -121,12 +124,22 @@ inline bool MethodDesc::IsLCGMethod()
     return ((mcDynamic == GetClassification()) && dac_cast<PTR_DynamicMethodDesc>(this)->IsLCGMethod());
 }
 
+// NOTE: This method only detects the subset of ILStubs that are internally represented using DynamicMethodDesc.
+// There are other methods compiled from IL generated at runtime via ILStubLinker that still return FALSE here.
+// See TryGenerateTransientILImplementation.
 inline bool MethodDesc::IsILStub()
 {
     WRAPPER_NO_CONTRACT;
     return ((mcDynamic == GetClassification()) && dac_cast<PTR_DynamicMethodDesc>(this)->IsILStub());
 }
 
+// This method is intended to identify runtime defined methods that don't have a corresponding Metadata or
```

Comment:
> identify runtime defined methods that don't have a corresponding Metadata or

I do not think this is accurate, e.g. async thunks do have associated metadata. They do have associated metadata token and module.

Reply:
Let me know if you have a suggestion on a better way to phrase this. I look at as:

In the user's source code they have some method "async Task Foo() { do stuff }"
At runtime we have both an AsyncThunk and the VariantImpl associated in some way with that metadata definition but since the user only defined one method we want to present them an abstraction that also only has one method. We picked VariantImpl as the one we'd show.

Reply:
I left suggestion in the next comment.